Use gdbm for NDBM_File.  The compatability header is named gdbm-ndbm.h
on debian machines.  Explicitly link against -lgdbm.

diff -ur perl-5.6.1.orig/Configure perl-5.6.1/Configure
--- perl-5.6.1.orig/Configure	Mon Mar 19 14:03:33 2001
+++ perl-5.6.1/Configure	Fri Jun 22 13:30:57 2001
@@ -14749,7 +14749,7 @@
 eval $inhdr
 
 : see if ndbm.h is available
-set ndbm.h t_ndbm
+set gdbm-ndbm.h t_ndbm
 eval $inhdr
 case "$t_ndbm" in
 $define)
diff -ur perl-5.6.1.orig/ext/NDBM_File/Makefile.PL perl-5.6.1/ext/NDBM_File/Makefile.PL
--- perl-5.6.1.orig/ext/NDBM_File/Makefile.PL	Fri Feb 23 13:57:54 2001
+++ perl-5.6.1/ext/NDBM_File/Makefile.PL	Fri Jun 22 13:31:36 2001
@@ -1,7 +1,7 @@
 use ExtUtils::MakeMaker;
 WriteMakefile(
     NAME	=> 'NDBM_File',
-    LIBS => ["-L/usr/local/lib -lndbm", "-ldbm -lucb"],
+    LIBS => ["-lgdbm"],
     MAN3PODS 	=> {}, 	# Pods will be built by installman.
     XSPROTOARG => '-noprototypes', 		# XXX remove later?
     VERSION_FROM => 'NDBM_File.pm',
diff -ur perl-5.6.1.orig/ext/NDBM_File/NDBM_File.xs perl-5.6.1/ext/NDBM_File/NDBM_File.xs
--- perl-5.6.1.orig/ext/NDBM_File/NDBM_File.xs	Sun Mar  4 06:53:20 2001
+++ perl-5.6.1/ext/NDBM_File/NDBM_File.xs	Fri Jun 22 13:28:44 2001
@@ -6,7 +6,7 @@
  * See also INSTALL section on DB3.
  * -- Stanislav Brabec <utx@penguin.cz> */
 #undef ENTER
-#include <ndbm.h>
+#include <gdbm-ndbm.h>
 
 typedef struct {
 	DBM * 	dbp ;
